examples/tutorial/index.js

```diff
@@ -0,0 +1,306 @@
+    const height = img.getAttribute('height');
+    let length = Futils.inlineTextLength(fnode.element) * 2;
+    if (Number.isNaN(length)) {
+        // Don't penalize empty captions
+        length = 1;
+    }
+    return {
+        score: width && height ? width * height / 100 : 100,
+        note: { length },
+    };
+}
+
+const hasAncestor = (tagName, scoreIfHas) => fnode => {
+    const lowerTag = tagName.toLowerCase();
+    for (let element = fnode.element, parent;
+        (parent = element.parentNode) !== null &&
+         parent.nodeType === parent.ELEMENT_NODE;
+        element = parent) {
+        if (element.tagName.toLowerCase() === lowerTag) {
+            return scoreIfHas;
+        }
+    }
+    return 1;
+};
+
+const rules = ruleset(
+    // Isolate the actual blog post body text. Based on Fathom's example
+    // Readability rules
+    rule(dom('p,li,ol,ul,code,blockquote,pre,h1,h2,h3,h4,h5,h6'),
+         props(scoreByLength).type('paragraphish')),
+    rule(type('paragraphish'), score(byInverseLinkDensity)),
+    rule(dom('p'), score(4.5).type('paragraphish')),
+
+    // Tweaks for this particular blog
+    rule(type('paragraphish'), score(hasAncestor('article', 10))),
+    rule(dom('.entry-summary p'), score(0).type('paragraphish')),
+    rule(dom('figure'), props(scoreByImageSize).type('paragraphish')),
+    rule(dom('.jetpack-video-wrapper'), props(() => ({
+        score: 100,
+        note: { length: 1 },
+    })).type('paragraphish')),
+
+    // Find the best cluster of paragraph-ish nodes
+    rule(
+        type('paragraphish').bestCluster({
+            splittingDistance: 3,
+            differentDepthCost: 6.5,
+            differentTagCost: 2,
+            sameTagCost: 0.5,
+            strideCost: 0,
+        }),
+        out('content').allThrough(Futils.domSort)));
+
+// FIXME: TODO: blog article, render template, custom SCSS
+class TutorialParser extends Libingester.HTMLArticleParser {
+    parseTitle ($) {
+        return super.parseTitle($).replace(/ - Creative Commons$/, '');
+    }
+
+    parseLastModifiedDate ($) {
+        // FIXME make it a metascraper rule
+        return $('meta[property="article:modified_time"]').attr('content');
+    }
+
+    parseLicense () {
+        return 'CC BY 4.0 International';
+    }
+
+    parseReadMoreText ($) {
+        // FIXME parsers dependencies
+        const title = this.parseTitle($);
+        const author = this.parseAuthors($);
+        return `"${title}" by ${author}, ` +
+            'used under CC BY 4.0 International / Reformatted from original';
+    }
+
+    parseTags ($) {
+        // Wordpress distinguishes predefined "categories" and free-form "tags".
+        // We are likely to make Wordpress categories into featured sets, and
+        // Wordpress tags non-featured. For now, we will mark the tag IDs of
+        // Wordpress tags with "tag:".
+        const wpCategory = $('meta[property="article:section"]')
+              .attr('content');
```

Comment:
Too much indentation

Reply:
According to? The lint command pass.

Reply:
I don't know, like I feel the indentation should be multiple of 4 in this case.

Reply:
OK I enabled indent MemberExpression in eslint for this.